src/vhip/mlmodel/gene_features.py

```diff
@@ -352,3 +354,38 @@ def RSCU(
                         synonymous_total_count / len(synonymous_codons)
                     )  # expected frequency of current codon given all assumption all synonymous codons are equally likely to encode the aa
                     self.RSCU_dict[codon] = self.codon_dict[codon] / expected_frequency
+
+
+class CodonBiasComparison:
+    """Class for calculating codon bias similarity between a virus GeneSet and a host GeneSet.
+
+    Args:
+        host_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a host GeneSet.
+        virus_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a virus GeneSet.
+    """
+
+    def __init__(
+        self,
+        host_dict: Union[dict[str, int], dict[str, float]],
+        virus_dict: Union[dict[str, int], dict[str, float]],
+    ) -> None:
+        """Initialize class variables and read in an annotated genes file, storing Gene objects and metadata in lists."""
+        self.host_dict = host_dict
+        self.host_list = list(self.host_dict.values())
+        self.virus_dict = virus_dict
+        self.virus_list = list(self.virus_dict.values())
+
+    def slope(self) -> None:
+        """Compute slope between host and virus codon bias using linear regression."""
+        self.slope = scipy.stats.linregress(self.host_list, self.virus_list)[0]
+
+    def R2(self) -> None:
+        """Compute R^2 value between host and virus codon bias using linear regression."""
+        r_value = scipy.stats.linregress(self.host_list, self.virus_list)[2]
+        self.R2 = r_value**2
+
```

Comment:
also there is an optimization step to consider. I see you ran the linregress function twice, one for the slope, and one for the R2. Instead I'd suggest running it once (make new method for that), and the slope and R2 methods extract the output of the linregress method. 

Let me know if that makes sense!

Reply:
Great idea. I optimized so the linregress function is only ran when it needs to be ran.